sharding/handler/log_handler.py

```diff
@@ -1,54 +1,42 @@
 import logging
 
-from sharding.handler.utils.web3_utils import (
-    get_recent_block_hashes,
-    get_canonical_chain,
-)
-
 
 class LogHandler:
 
-    logger = logging.getLogger("evm.chain.sharding.LogHandler")
-
-    def __init__(self, w3, history_size=256):
-        self.history_size = history_size
-        self.w3 = w3
-        # ----------> higher score
-        self.recent_block_hashes = get_recent_block_hashes(w3, history_size)
-
-    def get_new_logs(self, address=None, topics=None):
-        # TODO: should see if we need to do something with revoked_hashes
-        #       it seems reasonable to revoke logs in the blocks with hashes in `revoked_hashes`
-        revoked_hashes, new_block_hashes = get_canonical_chain(
-            self.w3,
-            self.recent_block_hashes,
-            self.history_size,
-        )
-        # determine `unchanged_block_hashes` by revoked_hashes
-        # Note: use if/else to avoid self.recent_block_hashes[:-1 * 0]
-        #       when len(revoked_hashes) == 0
-        if len(revoked_hashes) != 0:
-            unchanged_block_hashes = self.recent_block_hashes[:-1 * len(revoked_hashes)]
+    logger = logging.getLogger("sharding.handler.LogHandler")
+
+    def __init__(self, web3, period_length):
+        self.web3 = web3
+        self.period_length = period_length
+
+    def get_logs(
+            self,
+            *,
+            address=None,
+            topics=None,
+            from_block=None,
+            to_block=None):
+        current_block_number = self.web3.eth.blockNumber
+        if from_block is None:
+            # Search from the start of current period
+            fromBlock = current_block_number - current_block_number % self.period_length
         else:
-            unchanged_block_hashes = self.recent_block_hashes
-        # append new blocks to `unchanged_hashes`, and move revoked ones out of
-        # `self.recent_block_hashes`
-        new_recent_block_hashes = unchanged_block_hashes + new_block_hashes
-        # keep len(self.recent_block_hashes) <= self.history_size
-        self.recent_block_hashes = new_recent_block_hashes[-1 * self.history_size:]
-
-        if len(new_block_hashes) == 0:
-            return tuple()
-
-        from_block_hash = new_block_hashes[0]
-        to_block_hash = new_block_hashes[-1]
-        from_block_number = self.w3.eth.getBlock(from_block_hash)['number']
-        to_block_number = self.w3.eth.getBlock(to_block_hash)['number']
+            if from_block > current_block_number:
+                raise Exception(
+                    "Can not search logs starting with block number"
+                    "larger than current block number"
+                )
+            fromBlock = from_block
+        if to_block is None:
+            toBlock = current_block_number
```

Comment:
Can you explain why do we need new parameters `fromBlock` and `toBlock`? I know it's the original sin of `web3.js`, but still seems a little strange to see lower camel case here. Can we just use `from_block`, `to_block`?